mlflow/projects/backend/abstract_backend.py

```diff
@@ -0,0 +1,30 @@
+from abc import abstractmethod, ABCMeta
+
+
+class AbstractBackend():
+    """
+    Abstract class for MLflow Project Execution Backend
+    A backend must implement the run method to launch a new job
+    """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def run(self, run_id, experiment_id, project_uri, entry_point, params,
```

Comment:
Hm seems reasonable to allow the possibility of the custom backend (rather than the local Python client) creating a run, which will be returned indirectly via the `SubmittedRun` returned by `run`. The one advantage I see of the current approach is that it guarantees that the local client can access the run created for the project execution.

On the other hand, the current behavior of creating the run on the client puts the onus on backend implementors to guarantee that the backend (k8s, YARN, etc) has the appropriate credentials to log to the run. I could also see it being nice to be able to say "if you can authenticate to project backend X, you can run a project" & not have to worry about also configuring auth to a tracking server.

Regarding the conceptual relationship, in my mind it's more along the lines of "a run is a representation of project execution" than "project execution is a sub-operation of a run". That being said, I've seen use cases where users ran multiple projects within the same run, so that all the params for the different project executions would be surfaced within a single top-level run (i.e. in this case it's helpful to be able to say "backend X, please log to run 'abc'"). If backends had the flexibility to create their own run / ignore the user-supplied run ID to `mlflow.projects.run`, this use case would be difficult to achieve against those backends. 

After writing this out I'd still lean towards creating the run in the client / allowing the client to specify the run ID for the project execution, but definitely open to your alternative - were there other use cases/thoughts you had in mind to motivate this?

Reply:
@smurching I think you really nailed it with this explanation of the tradeoffs. In retrospect, I think my primary motivation for raising this was to determine whether or not we think the project backend and client should be tied to the same particular MLflow Tracking server. Given that the current structure of MLflow Projects operates under the assumption that both the client and the project executor are referring to Runs in the same location, I think it's fine to continue delegating run creation to the client and to stipulate that the Projects backend must communicate with the same MLflow Tracking server.